src/components/expression/expression-panel.js

```diff
@@ -224,16 +218,66 @@ genotet.ExpressionPanel.prototype.updateGenes = function(gene) {
       text: gene
     };
   });
-
   var profile = this.container.find('#profile select').empty();
-  this.selectProfiles_ = profile.select2({
-    data: genes,
-    multiple: true
-  });
-  this.container.find('#profile .select2-container').css({
-    width: '100%'
-  });
-  this.updateGeneSelections();
+
+  if (!genes.length) {
+    return;
+  }
+
+  $.fn.select2.amd.require(['select2/data/array', 'select2/utils'],
+    function(ArrayData, Utils) {
+      var CustomData = function($element, options) {
+        /**
+         * @type {{
+         *   __super__: function(?, ?)
+         * }}
+         */(CustomData).__super__.constructor.call(
+          this, $element, options);
+      };
+      /**
+       * @type {{
+       *   Extend:function()
+       * }}
+       */(Utils.Extend(CustomData, ArrayData));
+
+      var pageSize = genotet.ExpressionPanel.prototype.PROFILE_PAGE_SIZE_;
+      CustomData.prototype.query = function(params, callback) {
+        if (!('page' in params)) {
+          params.page = 1;
+        }
+        var data = {};
+        data.results = genes.slice((params.page - 1) * pageSize,
+          params.page * pageSize);
+        data.pagination = {};
+        data.pagination.more = params.page * pageSize < genes.length;
+        callback(data);
+      };
+
+      this.selectProfiles_ = profile.select2({
+        ajax: {},
```

Comment:
ack

On Saturday, March 19, 2016, Liana Yang <notifications@github.com> wrote:

> In src/components/expression/expression-panel.js
> <https://github.com/ViDA-NYU/genotet/pull/151#discussion_r56757944>:
>
> > +
> > +      var pageSize = genotet.ExpressionPanel.prototype.PROFILE_PAGE_SIZE_;
> > +      CustomData.prototype.query = function(params, callback) {
> > +        if (!('page' in params)) {
> > +          params.page = 1;
> > +        }
> > +        var data = {};
> > +        data.results = genes.slice((params.page - 1) * pageSize,
> > +          params.page * pageSize);
> > +        data.pagination = {};
> > +        data.pagination.more = params.page * pageSize < genes.length;
> > +        callback(data);
> > +      };
> > +
> > +      this.selectProfiles_ = profile.select2({
> > +        ajax: {},
>
> It is used to query data for select2 and it should be kept empty in this
> situation according to the document.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ViDA-NYU/genotet/pull/151/files/90437f8ff78facbc443fa2717a721540a82c9157#r56757944>
>